[App Config] Add sample for UpdateSyncToken

bbrusniak opened the pull request:
This fixes #16119.  I added two files, updateSyncTokenSample.ts and testData.ts, and modified the package.json in the appconfiguration folder and featureFlag.ts to fix a publishing issue. Also updated common/config/rush/pnpm-lock.yaml. Finally, republished samples in the samples/v1-beta/ folder. Here is what I specifically did for each file:

- updateSyncTokenSample.ts:
   - Contains a sample of decoding an event received from EventGrid, filtering for events specific to app configuration, and then logging the updated configurations. To reduce the complexity of the sample, updateSyncTokenSample uses a hard coded EventGridEvent instead of incorporating another service to act as a receiver.
- testData.ts
   - Contains a hard coded example of an EventGridEvent in string form.
- featureFlag.ts
   - Updated one of the operators to fix error while publishing. 
- package.json 
   - Added a devDepedency on EventGrid for this package.
- common/config/rush/pnpm-lock.yaml
   - Pushed changes created due to adding a dependency to EventGrid in the AppConfiguration package  
- samples/v1-beta/*
   - Republished samples  

Testing:

To test this sample, I created an app configuration resource on Azure, connected to it using the connection string, and verified that the sample behaved as expected. 

@jeremymeng if you could review as well that would be appreciated!

HarshaNalluru reviewed (approved):
Looks like a great translation of the .NET sample as expected in the issue.

jeremymeng commented:
@HarshaNalluru thanks for reviewing!
@bbrusniak We will need to publish the sample by following the https://github.com/Azure/azure-sdk-for-js/wiki/Samples-v2-Migration-Guide.  We also want to add this sample to the `skip` list in the package.json so it doesn't get picked up for automated testing.

HarshaNalluru commented:
Based on the build failures, it looks like something is wrong with the pnpm-lock file.
Can you 
- delete the pnpm-lock file in your branch
- copy the one in the main branch in the repository instead
And then run `rush update`?

bbrusniak commented:
@jeremymeng I added the sample to the "skip" list and published using the following commands.
```
$ rm -rf samples
$ rushx build
$ npx dev-tool samples publish
```
I got the following error: 
`samples-dev/featureFlag.ts:82:31 - error TS-AZURE: unsupported ES2020 "NullishCoalesce" syntax

82     clientFilter.parameters = clientFilter.parameters ?? {};

Suggestion: try `clientFilter.parameters || {}` if it does not affect runtime behavior, or use an explicit `if` block to handle the nullish case
`
I updated the featureFlag.ts file to use "||" instead of "??" and verified that the sample had the same output. 

I think everything should be good to go now

bbrusniak commented:
@HarshaNalluru I ran rush update locally and pushed the pnpm-lock which seems to have fixed the build issues.

jeremymeng commented:
> `samples-dev/featureFlag.ts:82:31 - error TS-AZURE: unsupported ES2020 "NullishCoalesce" syntax

Thanks for the fix @bbrusniak. Yes, we want the samples to run with lower versions too so new syntaxes are not supported.
